includes/sanitizers/class-amp-img-sanitizer.php

```diff
@@ -157,30 +157,29 @@ private function determine_dimensions( $need_dimensions ) {
 				if ( ! $node instanceof DOMElement ) {
 					continue;
 				}
+				$height = $dimensions && isset( $dimensions['height'] ) ? $dimensions['height'] : self::FALLBACK_HEIGHT;
+				$width  = $dimensions && isset( $dimensions['width'] ) ? $dimensions['width'] : self::FALLBACK_WIDTH;
+				$class  = $node->getAttribute( 'class' );
+				if ( ! $class ) {
+					$class = '';
+				}
 				if (
 					! is_numeric( $node->getAttribute( 'width' ) ) &&
 					! is_numeric( $node->getAttribute( 'height' ) )
 				) {
-					$height = self::FALLBACK_HEIGHT;
-					$width  = self::FALLBACK_WIDTH;
 					$node->setAttribute( 'width', $width );
 					$node->setAttribute( 'height', $height );
-					$class = $node->hasAttribute( 'class' ) ? $node->getAttribute( 'class' ) . ' amp-wp-unknown-size' : 'amp-wp-unknown-size';
-					$node->setAttribute( 'class', $class );
+					$node->setAttribute( 'class', trim( $class . ' amp-wp-unknown-size' ) );
```

Comment:
Is `amp-wp-unknown-size` even true here? Shouldn't it only be true if the `FALLBACK_HEIGHT` or `FALLBACK_WIDTH` are used? Otherwise, if we have `$dimensions` then we do know the size.